### PR TITLE
feat(source-scan): coerce `cargo-near` version in nested builds (for abi)

### PR DESCRIPTION
**Pull request by dj8yfo**
addresses [bug](https://github.com/near/cargo-near/pull/179#issuecomment-2226846711)
Host with CLI built from [cargo-near branch](https://github.com/dj8yfo/cargo-near/tree/artificially_bumped_version):
image

From [docker](https://github.com/dj8yfo/cargo-near-image/blob/0.13.0-fake-dev-cargo-near-git/Dockerfile#L36) from [cargo-near branch](https://github.com/dj8yfo/cargo-near/tree/artificially_bumped_version):
image

--- 

```bash
near contract call-function as-read-only donation-product.repro-fct-37.testnet contract_source_metadata json-args {} network-config testnet now
```

=> 

<details>
  <summary>donation-product.repro-fct-37.testnet</summary><p>
  
```json
{
  "build_info": {
    "build_command": [
      "cargo",
      "near",
      "build"
    ],
    "build_environment": "dj8yfo/sourcescan:0.x.x-dev-git-artif-0.13.0@sha256:bf0945f898cb141786df5e8ebfbaeefdd9f9c7afedabf3a1109b9d06743ca4bc",
    "contract_path": "product-donation",
    "source_code_snapshot": "git+https://github.com/dj8yfo/factory-rust?rev=0bc030fb4e3340f998b48e34f97823c3958c8cf7"
  },
  "link": "https://github.com/dj8yfo/factory-rust/tree/0bc030fb4e3340f998b48e34f97823c3958c8cf7",
  "standards": [
    {
      "standard": "nep330",
      "version": "1.2.0"
    }
  ],
  "version": "0.2.9"
}
```

</p></details>

--- 

```bash
near contract download-abi donation-product.repro-fct-37.testnet save-to-file donation-product.repro-fct-37.testnet.json network-config testnet now
```
=> 

```json
{
  "schema_version": "0.4.0",
  "metadata": {
    "name": "donation",
    "version": "0.2.9",
    "build": {
      "compiler": "rustc 1.79.0",
      "builder": "cargo-near 0.13.0"
    }
  },
  "body": {
  ...
```

--- 

```bash
sha256sum *.wasm
8d3b755c02f8b8341581af03b84c9608e59c557153d9fc64a3ba3c1b4c19a34e  donation-product.repro-fct-37.testnet.wasm
3a1c58f6d9fc0e3b0c987cd5e01fbf5c63627ade4babfd74fdf4ac7a566ab253  repro-fct-37.testnet.wasm
8d3b755c02f8b8341581af03b84c9608e59c557153d9fc64a3ba3c1b4c19a34e  repro-fct-product-37.testnet.wasm
```

### CODE REVIEW

File: cargo-near/src/build_extended/build_script.rs
Comment (frol) [inline cargo-near/src/build_extended/build_script.rs:107]: Please, make the warning more actionable, e.g.:

"WARNING: `cargo-near` crate version (potentially used in build.rs) did not match `cargo-near` build environment. Use the matching `cargo-near` CLI version or update `cargo-near` in Cargo.toml"

**Review by frol — APPROVED** [review]
@dj8yfo This makes sense. Thank you for preparing it!

Consider joining @race-of-sloths

**Comment by dj8yfo** [comment]
@race-of-sloths include

**Comment by race-of-sloths** [comment]
@dj8yfo Thank you for your contribution! Your pull request is now a part of the [Race of Sloths](https://race-of-sloths.com)!
Are you going to win race-of-sloths this month? If so, you should speed up! Run, Sloth, run!

[<picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=dark">
    <source media="(prefers-color-scheme: light)" srcset="https://badge.race-of-sloths.com/dj8yfo?type=bot&theme=light">
    Shows profile picture for the author of the PR
</picture>](https://race-of-sloths.com/profile/dj8yfo)


<details><summary>Current status: <i>waiting for scoring</i></summary>

We're waiting for maintainer to score this pull request with `@race-of-sloths score [0,1,2,3,5,8,13]` command
</details><details><summary>What is the Race of Sloths</summary>
<p>
Race of Sloths is a friendly competition where you can participate in challenges and compete with other open-source contributors within your normal workflow

For contributors:

- Tag @race-of-sloths inside your pull requests
- Wait for the maintainer to review and score your pull request
- Check out your position in the [Leaderboard](https://race-of-sloths.com/leaderboard)
- Keep weekly and monthly streaks to reach higher positions
- Boast your contributions with a dynamic picture of your [Profile](https://race-of-sloths.com/profile/dj8yfo)

For maintainers:
- Score pull requests that participate in the Race of Sloths
- Engage contributors with fair scoring and fast responses so they keep their streaks
- Promote the Race to the point where the Race starts promoting you
- Grow the community of your contributors

Feel free to check [our website](https://race-of-sloths.com) for additional details!
</p>
</details> 

<details>
<summary>Bot commands</summary>

- For contributors
  - **Include a PR:** `@race-of-sloths include` to enter the Race with your PR
- For maintainers:
  - **Assign points:** `@race-of-sloths score [1/2/3/5/8/13]` to award points based on your assessment.
  - **Reject this PR:** `@race-of-sloths exclude` to send this PR back to the drawing board.
  - **Exclude repo:** `@race-of-sloths pause` to stop bot activity in this repo until `@race-of-sloths unpause` command is called

</details>

**Review by frol — APPROVED** [review]

**Comment by frol** [comment]
@race-of-sloths score 8

**Comment by race-of-sloths** [comment]
### 🌟 Score recorded!

@frol, thank you for scoring this pull request in the Race of Sloths!

**Comment by race-of-sloths** [comment]
### ✅ PR is finalized! 
Your contribution is much appreciated with a final score of 8!
You have received 80 Sloth points for this contribution

**Pull request merged**
